capabilities/batch-processing/batch-processing-policy.xml

```diff
@@ -1,13 +1,127 @@
+        <set-variable name="rp10s-limit" value="@{
+            JObject selectedDeployment = (JObject)context.Variables["selected-deployment"];
+            return selectedDeployment.Value<int>("rp10s-limit");
+        }" />
+        <!-- calculate consumed tokens from request body -->
+        <set-variable name="consumed-tokens" value="@{
+            JObject requestBody = context.Request.Body.As<JObject>(preserveContent: true);
+            if(requestBody.ContainsKey("max_tokens"))
+            {
+                return requestBody.Value<int>("max_tokens");
+            }
+            else
+            {
+                return 16;
+            }
+        }" />
+        <!-- apply tpm and rp10s limits for the model deployment -->
+        <rate-limit-by-key counter-key="@(context.Variables["selected-deployment-id"] + "|tokens-limit")"
+            calls="@((int)context.Variables["tpm-limit"])"
+            renewal-period="60"
+            increment-count="@((int)context.Variables["consumed-tokens"])"
+            retry-after-header-name="x-apim-tokens-retry-after"
+            retry-after-variable-name="tokens-retry-after"
+            remaining-calls-header-name="x-apim-remaining-tokens" 
+            remaining-calls-variable-name="remaining-tokens"
+            total-calls-header-name="x-apim-total-tokens"/>
+        <rate-limit-by-key counter-key="@(context.Variables["selected-deployment-id"] + "|requests-limit")"
+            calls="@((int)context.Variables["rp10s-limit"])"
+            renewal-period="10"
+            retry-after-header-name="x-apim-requests-retry-after"
+            retry-after-variable-name="requests-retry-after"
+            remaining-calls-header-name="x-apim-remaining-requests"
+            remaining-calls-variable-name="remaining-requests"
+            total-calls-header-name="x-apim-total-requests"/>
+        <!-- determine if the request is a batch request -->
+        <set-variable name="is-batch" value="@(bool.Parse(context.Request.Url.Query.GetValueOrDefault("is-batch", "false")))" />
+        <choose>
+            <when condition="@((bool)context.Variables["is-batch"])">
+                <!-- retrieve batch tpm and rp10s limits for the model deployment -->
+                <set-variable name="batch-tpm-threshold" value="@{
+                    JObject selectedDeployment = (JObject)context.Variables["selected-deployment"];
+                    return selectedDeployment.Value<int>("batch-tpm-threshold");
+                }" />
+                <set-variable name="batch-rp10s-threshold" value="@{
+                    JObject selectedDeployment = (JObject)context.Variables["selected-deployment"];
+                    return selectedDeployment.Value<int>("batch-rp10s-threshold");
+                }" />
+                <!-- apply batch tpm and rp10s limits for the model deployment -->
+                <rate-limit-by-key counter-key="@(context.Variables["selected-deployment-id"] + "|batch-tokens-limit")"
```

Comment:
In my policy, the `batch-tpm-threshold` is the level of remaining tokens that must be available to allow a batch request through.

I think this implementation will have a different behaviour?

That said, I'm happy to merge it as-is - we're likely to have various tweaks needed to each approach as we start the load testing!